src/libnmstate/schemas/interfaces.json

```diff
@@ -0,0 +1,30 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+
+    "title": "Interfaces tree",
+    "type": "object",
+    "properties": {
+        "interfaces": {
+            "title": "Interfaces",
+            "type": "array",
+            "items": {
+                "title": "Interface",
+                "type": "object",
+                "properties": {
+                    "name": { "type": "string" },
+                    "description": { "type": "string" },
+                    "type": {
+                        "type": "string",
+                        "enum": ["ethernet", "bond", "vlan"]
```

Comment:
type should be required too

Reply:
Nope, it actually does not need to be required: Given an interface, a user may want to change its MTU without knowing of what type it is. Sounds reasonable (for now at least) to me.
Even true for removing an interface.
Anyway, the discussion fits the next PR.